src/modules/fancyzones/FancyZonesLib/Layout.cpp

```diff
@@ -108,15 +108,20 @@ Layout::Layout(const LayoutData& data) :
 
 bool Layout::Init(const FancyZonesUtils::Rect& workArea, HMONITOR monitor) noexcept
 {
-    //invalid work area
+    // invalid work area
     if (workArea.width() == 0 || workArea.height() == 0)
     {
         Logger::error(L"Layout initialization: invalid work area");
         return false;
     }
 
-    //invalid zoneCount, may cause division by zero
-    if (m_data.zoneCount <= 0 && m_data.type != FancyZonesDataTypes::ZoneSetLayoutType::Custom)
+    // invalid zoneCount, may cause division by zero
+    bool isGridType = m_data.type == FancyZonesDataTypes::ZoneSetLayoutType::Columns 
```

Comment:
what about custom grid type? Should it be here?

Reply:
This is not possible because of the editor, and if the file is changed manually the layout will be marked as invalid and won't be calculated.
So it could make sense only for readability or in favor of future changes.